src/iris/bin/app_stats.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+# See LICENSE in the project root for license information.
+
+from gevent import monkey, sleep, spawn
+monkey.patch_all()  # NOQA
+
+import logging
+import os
+
+from iris import db, metrics
+from iris.api import load_config, ApplicationStats
+
+# logging
+logger = logging.getLogger()
+formatter = logging.Formatter('%(asctime)s %(levelname)s %(name)s %(message)s')
+log_file = os.environ.get('APP_STATS_LOG_FILE')
+if log_file:
+    ch = logging.handlers.RotatingFileHandler(log_file, mode='a', maxBytes=10485760, backupCount=10)
+else:
+    ch = logging.StreamHandler()
+ch.setLevel(logging.DEBUG)
+ch.setFormatter(formatter)
+logger.setLevel(logging.INFO)
+logger.addHandler(ch)
+
+
+# pidfile
+pidfile = os.environ.get('APP_STATS_PIDFILE')
+if pidfile:
+    try:
+        pid = os.getpid()
+        with open(pidfile, 'w') as h:
+            h.write('%s\n' % pid)
+            logger.info('Wrote pid %s to %s', pid, pidfile)
+    except IOError:
+        logger.exception('Failed writing pid to %s', pidfile)
+
+stats_reset = {
+    'task_failure': 0,
+}
+
+
+def set_app_stats(app, stats, connection, cursor):
+    for stat, val in stats.iteritems():
+        cursor.execute('''REPLACE INTO `application_stats` (`application_id`, `statistic`, `value`, `timestamp`)
```

Comment:
Instead of doing `REPLACE INTO`, can you refactor this to use `INSERT INTO ... ON DUPLICATE KEY UPDATE ..` ?